### PR TITLE
fix: duplicate preview project bugfix and types

### PR DESCRIPTION
**Pull request by rephus**
Closes:  #9617

### Description:
Small improvements and bugfixes 

Screenshot from 2024-04-03 10-28-16

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/models/ProjectModel/ProjectModel.ts [hunk 1]
```diff
@@ -1102,7 +1103,17 @@ export class ProjectModel {
                     ? await trx('spaces')
                           .insert(
                               dbSpaces.map((d) => {
-                                  const createSpace = {
+                                  type CloneSpace = Omit<
```
Comment (rephus) [inline packages/backend/src/models/ProjectModel/ProjectModel.ts:1106]: added some types to ensure all mandatory fields are present, remove `autogenerated` fields from the type.

File: packages/backend/src/models/ProjectModel/ProjectModel.ts [hunk 2]
```diff
@@ -1111,6 +1122,7 @@ export class ProjectModel {
                                   };
                                   // Remove the keys for the autogenerated fields
                                   // Some databases do not support undefined values
+                                  delete createSpace.search_vector;
```
Comment (rephus) [inline packages/backend/src/models/ProjectModel/ProjectModel.ts:1125]: this is autogenerated

File: packages/backend/src/models/ProjectModel/ProjectModel.ts [hunk 3]
```diff
@@ -1454,29 +1483,39 @@ export class ProjectModel {
...
-                    })
-                    .where('saved_query_id', chart.saved_query_id);
-            });
+            const updateChartInDashboards = newChartsInDashboards.map(
+                (chart) => {
+                    const newDashboardUuid = dashboardMapping.find(
+                        (m) => m.uuid === chart.dashboard_uuid,
+                    )?.newUuid;
+
+                    if (!newDashboardUuid) {
```
Comment (rephus) [inline packages/backend/src/models/ProjectModel/ProjectModel.ts:1504]: Fixes this bug 

Screenshot from 2024-04-03 09-29-01

**Comment by rephus** [comment]
I was unable to replicate this issue https://github.com/lightdash/lightdash/issues/9617 though

**Review by joaoviana — APPROVED** [review]

**Pull request merged**